byteutils.nim

```diff
@@ -84,13 +88,13 @@ proc toHexAux(ba: openarray[byte]): string {.noSideEffect.} =
     result[2*i] = hexChars[int ba[i] shr 4 and 0xF]
     result[2*i+1] = hexChars[int ba[i] and 0xF]
 
-proc toHex*(ba: openarray[byte]): string {.noSideEffect, inline.} =
+func toHex*(ba: openarray[byte]): string {.inline.} =
   ## Convert a byte-array to its hex representation
   ## Output is in lowercase
   ## No "endianness" reordering is done.
   toHexAux(ba)
 
-proc toHex*[N: static[int]](ba: array[N, byte]): string {.noSideEffect, inline.} =
+func toHex*[N: static[int]](ba: array[N, byte]): string {.inline.} =
```

Comment:
Do we need separate funcs for static/openarray? The only difference at code gen is the array one passes the length as a static.

Reply:
No we don't but if I remember correctly either:
  - I couldn't compile the code.
  - or I wanted to have more optimized code for arrays since length is static.